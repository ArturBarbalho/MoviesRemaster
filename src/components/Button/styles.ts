import styled, { css } from 'styled-components'
type props = {
    color:'gray'|'black'
}
export const Wrapper = styled.button<props>`
${({color})=>css`
background-color:${color};
color: white;
padding: 10px;
border-radius:20px;
margin-top:30px;
border:none;
width:150px;
font-size:larger;
:Hover{
    cursor:pointer;
    background-color:white;
    color:black;
    opacity:0.8;
}
`}

`