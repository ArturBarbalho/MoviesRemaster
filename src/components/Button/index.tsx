import { MouseEventHandler } from 'react'
import * as S from './styles'
type props = {
  onClick:MouseEventHandler<HTMLButtonElement>
  color:'gray'|'black'
}
const Button = ({onClick, color}:props) => (
  <S.Wrapper color={color} onClick={onClick} >
   See more
  </S.Wrapper>
)

export default Button