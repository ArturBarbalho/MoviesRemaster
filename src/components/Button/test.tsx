import { render, screen } from '@testing-library/react'
import userEvent from '@testing-library/user-event' 
import Button from '.'

describe('<Button />', () => {
  const onclick = jest.fn()
  it('should render the black color and execute the onclick function', () => {
    const { container } = render(<Button color='black' onClick={ ()=> onclick()} />)

    expect(onclick).not.toBeCalled()

    userEvent.click(screen.getByRole('button', { name: /See more/i }))

    expect(onclick).toBeCalled()  


    expect(screen.getByRole('button', { name: /See more/i })).toHaveStyle(
      'background-color:black;'
    )
    expect(container.firstChild).toMatchSnapshot()
  })


  it('should render the gray color', () => {
    const { container } = render(<Button color='gray' onClick={()=> onclick()}/>)

    expect(screen.getByRole('button', { name: /See more/i })).toHaveStyle(
      'background-color:gray;'
    )

    expect(container.firstChild).toMatchSnapshot()
  })
})