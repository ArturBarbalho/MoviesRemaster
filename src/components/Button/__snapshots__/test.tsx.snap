// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should render the black color and execute the onclick function 1`] = `
<button
  class="styles__Wrapper-sc-11qq8g6-0 hNDJSY"
  color="black"
>
  See more
</button>
`;

exports[`<Button /> should render the gray color 1`] = `
<button
  class="styles__Wrapper-sc-11qq8g6-0 jDrFTe"
  color="gray"
>
  See more
</button>
`;

exports[`<Button /> should render the heading 1`] = `
<button
  class="styles__Wrapper-sc-11qq8g6-0 jDrFTe"
  color="gray"
>
  See more
</button>
`;
